src/components/header.js

```diff
@@ -32,29 +34,90 @@ function Header({ slug }) {
   const [isOpen1, setIsOpen1] = useState(false);
   const [isOpen2, setIsOpen2] = useState(false);
 
-  /* first dropdown items */
-  const testMenuItemsOne = [
-    <Link to="/link-one" key="one">
-      {t("nav.dropdownOne.simpleLinkOne")}
-    </Link>,
-    <Link to="/link-two" key="two">
-      {t("nav.dropdownOne.simpleLinkTwo")}
-    </Link>,
-  ];
-
-  /* second dropdown items */
-  const testMenuItemsTwo = [
-    <Link to="/link-three" key="one">
-      {t("nav.dropdownTwo.simpleLinkThree")}
-    </Link>,
-    <Link to="/link-four" key="two">
-      {t("nav.dropdownTwo.simpleLinkFour")}
-    </Link>,
-  ];
-
-  /* links above search button */
-  const secondaryLinks = [];
+  /* dynamically creating dropdowns in navigation bar */
+  let dropdowns = [];
+  const dropdownLinks = links.navDropdowns;
+  const constFileLength = dropdownLinks.length;
+  const dropdownLabels = t("header.nav.dropdowns");
+  const translationFileLength = dropdownLabels.length;
+  const length =
+    translationFileLength > constFileLength
+      ? constFileLength
+      : translationFileLength; // take shorter length in case there is missing dropdowns in `../constants/links.js` (constants file) or `../locales/` (translation files)
+  if (translationFileLength !== constFileLength)
```

Comment:
Nice adding the warning! Although, I was wondering if it should be an error instead of a warning?

Reply:
Good idea. After all, they are going to get unexpected behavior. I didn't want to give an error message at first because I thought it would crash the whole site, but after adding `console.error`, I see it still loads the whole page, but with a different type of message in the console.